src/solvers/scip.rs

```diff
@@ -259,4 +396,62 @@ mod tests {
         let solution = model.solve().unwrap();
         assert_eq!((solution.value(x), solution.value(y)), (2., 0.));
     }
+
+    #[test]
+    fn can_set_int_param() {
+        let mut vars = variables!();
+        let x = vars.add(variable().clamp(0, 1));
+        let model = vars
+            .maximise(x)
+            .using(scip)
+            .set_option("display/verblevel", 2);
+        assert_eq!(model.get_option::<i32>("display/verblevel"), 2);
+    }
+
+    #[test]
+    fn can_set_real_param() {
+        let mut vars = variables!();
+        let x = vars.add(variable().clamp(0, 1));
+        let model = vars.maximise(x).using(scip).set_option("limits/time", 0.);
+        assert_eq!(model.get_option::<f64>("limits/time"), 0.);
+    }
+
+    #[test]
+    fn can_set_str_param() {
+        let mut vars = variables!();
+        let x = vars.add(variable().clamp(0, 1));
+        let model = vars
+            .maximise(x)
+            .using(scip)
+            .set_option("concurrent/paramsetprefix", "custom/path");
+        assert_eq!(
+            &model.get_option::<String>("concurrent/paramsetprefix"),
+            "custom/path"
+        );
+    }
+
+    #[test]
+    fn can_set_longint_param() {
+        let mut vars = variables!();
+        let x = vars.add(variable().clamp(0, 1));
+        let model = vars
+            .maximise(x)
+            .using(scip)
+            .set_option::<i64>("constraints/components/nodelimit", 99);
+        assert_eq!(
+            model.get_option::<i64>("constraints/components/nodelimit"),
+            99
+        );
+    }
+
+    #[test]
+    fn can_set_bool_param() {
+        let mut vars = variables!();
+        let x = vars.add(variable().clamp(0, 1));
+        let model = vars
+            .maximise(x)
+            .using(scip)
+            .set_option("display/allviols", true);
+        assert_eq!(model.get_option::<bool>("display/allviols"), true);
+    }
```

Comment:
None of these tests pass. The returned values are the defaults, so it seems like setting the values has no effect. @mmghannam do you spot anything obvious that I'm missing?

Reply:
I tried both the bool and the longint tests in russcip and they both work. I don't see anything you're missing though.

Reply:
Hmmm then I'll have to debug why these tests fail, thanks

Reply:
I tried this branch in my project and something very fishy is going on … it works well as long as I don't use any of the options introduced in this pull request. However, a mere call to `set_verbosity` causes the MILP to be solved in 0 seconds, followed by a segmentation fault. This is fun

Reply:
Learnt something today, nice, all fixed